### ISSUE TITLE
Pretty URLs Breaks Dark Mode on Apache

### ISSUE THREAD

**Issue by SebSeifert** [nextcloud/server#11884]
### Steps to reproduce
1. Turn Pretty URLs on.
2. Activate Dark Mode

### Expected behaviour
Dark Mode Turns On.

### Actual behaviour
Theme doesn't show the dark mode and for Fonts and High Contrast the Same.
If i disable the Pretty URL the Dark Mode Works.
### Server configuration

**Operating system**:
Ubuntu 18.04
**Web server:**
Apache2
**Database:**
MariaDB
**PHP version:**
7.2
**Nextcloud version:** (see Nextcloud admin page)
14.0.3
**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated
**Where did you install Nextcloud from:**

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.0.1
  - activity: 2.7.0
  - admin_audit: 1.4.0
  - calendar: 1.6.2
  - cloud_federation_api: 0.0.1
  - comments: 1.4.0
  - contacts: 2.1.6
  - dav: 1.6.0
  - federatedfilesharing: 1.4.0
  - federation: 1.4.0
  - files: 1.9.0
  - files_pdfviewer: 1.3.2
  - files_sharing: 1.6.2
  - files_texteditor: 2.6.0
  - files_trashbin: 1.4.1
  - files_versions: 1.7.1
  - files_videoplayer: 1.3.0
  - gallery: 18.1.0
  - gpxedit: 0.0.9
  - gpxmotion: 0.0.7
  - gpxpod: 2.3.1
  - logreader: 2.0.0
  - lookup_server_connector: 1.2.0
  - mail: 0.11.0
  - notifications: 2.2.1
  - oauth2: 1.2.1
  - password_policy: 1.4.0
  - provisioning_api: 1.4.0
  - richdocuments: 2.0.13
  - serverinfo: 1.4.0
  - sharebymail: 1.4.0
  - support: 1.0.0
  - survey_client: 1.2.0
  - theming: 1.5.0
  - twofactor_backupcodes: 1.3.1
  - updatenotification: 1.4.1
  - workflowengine: 1.4.0
Disabled:
  - encryption
  - files_external
  - firstrunwizard
  - nextcloud_announcements
  - systemtags
  - user_external
  - user_ldap
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
In this config the Pretty URLs are deactivated.

{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.seifert-online.eu",
            "bitwarden.seifert-online.eu"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/cloud.seifert-online.eu\/",
        "dbtype": "mysql",
        "version": "14.0.3.0",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "theme": "",
        "loglevel": 2,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "maintenance": false,
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpsecure": "ssl",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***"
    }
}
```
</details>

**Are you using external storage, if yes which one:** smb

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**
Chrome and Firefox
**Operating system:**
Windows 10
#### Web server error log
<details>
<summary>Web server error log</summary>

```
Shows nothing related to the problem.
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Shows nothing related to the problem.
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
When i press on Dark Mode in Accessibility

{
  "log": {
    "version": "1.1",
    "creator": {
      "name": "Firefox",
      "version": "62.0"
    },
    "browser": {
      "name": "Firefox",
      "version": "62.0"
    },
    "pages": [
      {
        "startedDateTime": "2018-10-17T14:12:38.928+02:00",
        "id": "page_1",
        "title": "Einstellungen - Seifert's Nextcloud",
        "pageTimings": {
          "onContentLoad": -1833,
          "onLoad": -1298
        }
      }
    ],
    "entries": [
      {
        "pageref": "page_1",
        "startedDateTime": "2018-10-17T14:12:38.928+02:00",
        "request": {
          "bodySize": 15,
          "method": "POST",
          "url": "https://cloud.seifert-online.eu/ocs/v2.php/apps/accessibility/api/v1/config/theme",
          "httpVersion": "HTTP/2.0",
          "headers": [
            {
              "name": "Host",
              "value": "cloud.seifert-online.eu"
            },
            {
              "name": "User-Agent",
              "value": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:62.0) Gecko/20100101 Firefox/62.0"
            },
            {
              "name": "Accept",
              "value": "application/json, text/plain, */*"
            },
            {
              "name": "Accept-Language",
              "value": "de,en-US;q=0.7,en;q=0.3"
            },
            {
              "name": "Accept-Encoding",
              "value": "gzip, deflate, br"
            },
            {
              "name": "Content-Type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "Content-Length",
              "value": "15"
            },
            {
              "name": "Cookie",
              "value": "**REMOVED**"
            },
            {
              "name": "Connection",
              "value": "keep-alive"
            },
            {
              "name": "TE",
              "value": "Trailers"
            }
          ],
          "cookies": [
            {
              "name": "__Host-nc_sameSiteCookielax",
              "value": "true"
            },
            {
              "name": "__Host-nc_sameSiteCookiestrict",
              "value": "true"
            },
            {
              "name": "ajs_user_id",
              "value": "null"
            },
            {
              "name": "ajs_group_id",
              "value": "null"
            },
            {
              "name": "ajs_anonymous_id",
              "value": "\"00000000000000000000000000\""
            },
            {
              "name": "nc_username",
              "value": "Administrator"
            },
          ],
          "queryString": [],
          "headersSize": 990,
          "postData": {
            "mimeType": "application/json;charset=utf-8",
            "params": [],
            "text": "{\"value\":false}"
          }
        },
        "response": {
          "status": 200,
          "statusText": "OK",
          "httpVersion": "HTTP/2.0",
          "headers": [
            {
              "name": "date",
              "value": "Wed, 17 Oct 2018 12:12:39 GMT"
            },
            {
              "name": "server",
              "value": "Apache/2.4.29 (Ubuntu)"
            },
            {
              "name": "expires",
              "value": "Thu, 19 Nov 1981 08:52:00 GMT"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none';base-uri 'none';manifest-src 'self'"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=15552000; includeSubDomains"
            },
            {
              "name": "content-length",
              "value": "74"
            },
            {
              "name": "referrer-policy",
              "value": "no-referrer-when-downgrade"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "x-robots-tag",
              "value": "none"
            },
            {
              "name": "x-download-options",
              "value": "noopen"
            },
            {
              "name": "x-permitted-cross-domain-policies",
              "value": "none"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "X-Firefox-Spdy",
              "value": "h2"
            }
          ],
          "cookies": [],
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 74,
            "text": "{\"ocs\":{\"meta\":{\"status\":\"ok\",\"statuscode\":200,\"message\":\"OK\"},\"data\":[]}}"
          },
          "redirectURL": "",
          "headersSize": 659,
          "bodySize": 733
        },
        "cache": {},
        "timings": {
          "blocked": 0,
          "dns": 0,
          "connect": 0,
          "ssl": 0,
          "send": 0,
          "wait": 115,
          "receive": 0
        },
        "time": 115,
        "_securityState": "secure",
        "serverIPAddress": "2a01:4f8:1c0c:7993::1",
        "connection": "443"
      },
      {
        "pageref": "page_1",
        "startedDateTime": "2018-10-17T14:12:39.057+02:00",
        "request": {
          "bodySize": 0,
          "method": "GET",
          "url": "https://cloud.seifert-online.eu/apps/accessibility/css/user-f484570d7cf557020e11ace406901b10.css?v=1539778359050",
          "httpVersion": "HTTP/2.0",
          "headers": [
            {
              "name": "Host",
              "value": "cloud.seifert-online.eu"
            },
            {
              "name": "User-Agent",
              "value": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:62.0) Gecko/20100101 Firefox/62.0"
            },
            {
              "name": "Accept",
              "value": "text/css,*/*;q=0.1"
            },
            {
              "name": "Accept-Language",
              "value": "de,en-US;q=0.7,en;q=0.3"
            },
            {
              "name": "Accept-Encoding",
              "value": "gzip, deflate, br"
            },
            {
              "name": "Connection",
              "value": "keep-alive"
            },
            {
              "name": "TE",
              "value": "Trailers"
            }
          ],
          "cookies": [
            {
              "name": "__Host-nc_sameSiteCookielax",
              "value": "true"
            },
            {
              "name": "__Host-nc_sameSiteCookiestrict",
              "value": "true"
            },
            {
              "name": "ajs_user_id",
              "value": "null"
            },
            {
              "name": "ajs_group_id",
              "value": "null"
            },
            {
              "name": "ajs_anonymous_id",
              "value": "\"00000000000000000000000000\""
            },
          ],
          "queryString": [
            {
              "name": "v",
              "value": "1539778359050"
            }
          ],
          "headersSize": 834
        },
        "response": {
          "status": 404,
          "statusText": "Not Found",
          "httpVersion": "HTTP/2.0",
          "headers": [
            {
              "name": "date",
              "value": "Wed, 17 Oct 2018 12:12:39 GMT"
            },
            {
              "name": "server",
              "value": "Apache/2.4.29 (Ubuntu)"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "cache-control",
              "value": "max-age=31536000, immutable"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none';base-uri 'none';manifest-src 'self';script-src 'nonce-YkJzcWY1RGppWkVtTEVRUFMydVhGWmtFS0h6anphRmM0K2NPcTJqY1RZND06RkZwNlR1alIrZlZuWEJWY0NTSGFXY00zZUUyRS8rY21tYUpqNTFHcUc3ND0=' 'unsafe-eval';style-src 'self' 'unsafe-inline';img-src 'self' data: blob:;font-src 'self';connect-src 'self';media-src 'self';frame-src https://office.seifert-online.eu"
            },
            {
              "name": "content-disposition",
              "value": "inline; filename=\"\""
            },
            {
              "name": "expires",
              "value": "Thu, 17 Oct 2019 12:12:39 +0000"
            },
            {
              "name": "pragma",
              "value": "cache"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=15552000; includeSubDomains"
            },
            {
              "name": "content-length",
              "value": "0"
            },
            {
              "name": "referrer-policy",
              "value": "no-referrer-when-downgrade"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "x-robots-tag",
              "value": "none"
            },
            {
              "name": "x-download-options",
              "value": "noopen"
            },
            {
              "name": "x-permitted-cross-domain-policies",
              "value": "none"
            },
            {
              "name": "content-type",
              "value": "text/css;charset=UTF-8"
            },
            {
              "name": "X-Firefox-Spdy",
              "value": "h2"
            }
          ],
          "cookies": [],
          "content": {
            "mimeType": "text/css; charset=UTF-8",
            "size": 0,
            "text": ""
          },
          "redirectURL": "",
          "headersSize": 998,
          "bodySize": 998
        },
        "cache": {},
        "timings": {
          "blocked": 0,
          "dns": 0,
          "connect": 0,
          "ssl": 0,
          "send": 0,
          "wait": 78,
          "receive": 0
        },
        "time": 78,
        "_securityState": "secure",
        "serverIPAddress": "2a01:4f8:1c0c:7993::1",
        "connection": "443"
      }
    ]
  }
}
```
</details>

**Comment by juliushaertl** [comment]
@skjnldsv According to the logs`/apps/accessibility/css/user-f484570d7cf557020e11ace406901b10.css?v=1539778359050` returns a 404. Any idea why that might happen?

**Comment by skjnldsv** [comment]
My guess is a wrong redirect setting on the Apache pretty url config?
But theoretically we should use index.php for this kind of url.

@SebSeifert if you access `index.php/apps/accessibility/css/user-f484570d7cf557020e11ace406901b10.css?v=1539778359050` does it work?

**Comment by MorrisJobke** [comment]
Fix is in #13786
